### DISCUSSION TITLE
Adding custom domain to gh-pages branch resets after each re-deploy

### DISCUSSION THREAD

**Discussion by Noxirus** [community/community#38250] [Other Feature Feedback, Questions, & Ideas] [answered]
I am working on a website project and trying to connect to my custom domain. One thing I want to know is, for the homepage string in the package.json does it need to be the custom domain name? Or does it need to be the usual: username.github.io/appname.

In this case should homepage be: https://noxirus.github.io/hamlicks-fresh-scoop/ OR www.hamlicks.com (my custom domain).

Any help would be appreciated, when I view the site at: https://noxirus.github.io/hamlicks-fresh-scoop/ it displays correctly, but if I add my custom domain it responds 404.

Thanks very much for your time.

-Noxirus

**Comment by DSDmark** [comment, marked as answer]
Hi @Noxirus,

I'm not sure, but you could try this:-

in package.json file write normal URL which is GitHub by default set like ```"homepage": "https://username.github.io/repository-name"``` in your case its ```https://noxirus.github.io/hamlicks-fresh-scoop/```

then if you want to add custom domain name then you can add simply inside the settings like this:-

![Screenshot 2022-11-04 at 12-47-19 DSDmark_TodoList The question “How do I build my own First React Web app” is a difficult one It seems like everyone has their own answers This Application tell that how to build](https://user-images.githubusercontent.com/85744816/199915007-7e03e542-4147-4f04-aeb7-d18b92de4c4d.png)

Then I think it's redirect to your custom URL.

And I think you should check this out:-
[**deploying site with custom URL**](https://medium.com/@madelinecorman/deploying-your-react-app-with-github-pages-and-adding-a-godaddy-custom-domain-name-50a99accd455)

**Reply by Noxirus** [reply]
Thanks for the support! Unfortunately when I set the homepage to https://noxirus.github.io/hamlicks-fresh-scoop/, assign my custom domain, and THEN npm run deploy, it forgets my custom domain and reverts it back to the https://noxirus.github.io/hamlicks-fresh-scoop/ URL. Addendum: It also reverts it back to this URL if I set the homepage URL to my custom domain as well.

The DNS seems to be connected correctly, when I connect the custom domain it verifies it and allows for HTTPS enforcement.

Not sure what branch I should be referring pages to as well. Some sources say gh-pages branch, but deploying my changes on that one is immediately removing my custom domain. Currently both branches are also responding with 404 (in the console, its a blank white page when attempting to access the site).

Any further support would be great!

-Noxirus

**Reply by Noxirus** [reply]
Addendum: When I change the branch to the main one for deployment reference, if I update/(push to the repo) it triggers that it has been deployed and updated recently, does NOT change my custom domain back to the https://noxirus.github.io/hamlicks-fresh-scoop/. 

I stopped ignoring the Build folder thinking that maybe if the build folder is in the repo the deployment would reference this particular folder when you are viewing the site, but the README is still shown instead.

**Reply by Noxirus** [reply]
Update: I learned I need to refer to the build folder that is created when deploying to the gh-pages, the issue here though is that when I use my custom domain with this particular branch it immediately overrides the custom domain, when I put the custom domain back in it no longer finds my site and re-deploying to the gh-pages branch erases my custom domain reference again.

**Reply by Noxirus** [reply]
Update: The issue my site was not being found is because of naming conventions, my custom domain I needed to ensure I was adding the www to the beginning of the address. Linking it to the gh-pages branch has made it so you can now see the website.

It's working which is nice, but there still is the small issue that if I run the npm run deploy command it will remove the refence to my custom domain on the pages section. I have to re-add it everytime I am deploying to my live site.

**Reply by Noxirus** [reply]
Update: The domain being removed has been resolved, you can find that resolution in this article here: https://github.com/tschaub/gh-pages/issues/213.

**Comment by Noxirus** [comment]
Hi @DSDmark 

Thanks again for the help, going through the article helped me get back on track with the necessary things that needed to be in place. The homepage URL needed to be: https://noxirus.github.io WITHOUT the repo name on the end, the custom domain was pointing to something that wasn't there.

Appreciate the help

-Noxirus
